frappe/frappe
# fix: User Account Deletion Flow

### Web Form Changes

The webform for account deletion has been renamed to **Request for Account Deletion** to clearly indicate its scope.

### Account Deletion Settings

Account Deletion Settings have been added to Website Settings. 
Here admins can enable the following:
  - If the form link should be visible on My Account Page
  - Number of days within which the account will be auto-deleted

<img width="1083" alt="Screenshot 2021-12-16 at 12 17 55 PM" src="https://user-images.githubusercontent.com/31363128/146322055-cd31e629-fb72-4f18-8d18-9568db62f12b.png">

### Description change

The description now indicates that both user's account, as well as their data, will be deleted if they go ahead with this form. The number of days within which the account will be deleted is also displayed.

<img width="840" alt="Screenshot 2021-12-01 at 8 53 14 PM" src="https://user-images.githubusercontent.com/31363128/144262261-bb023e57-a34d-4636-aea1-4d4a58b5eeca.png">

### On Hold status

The system admins can now put a request on hold if they find some issues with the request.

<img width="1303" alt="Screenshot 2021-12-16 at 12 21 19 PM" src="https://user-images.githubusercontent.com/31363128/146322371-0db71ebf-b23b-4c02-aee5-1e77e00d691e.png">

Put on Hold button will be displayed if the user is a System Manager and the current status is Pending Approval.

### Auto Deletion

A daily scheduled job has been added which will auto-delete such user accounts if the System Manager doesn't act within the days specified in settings. Only requests with status Pending Approval will be considered for auto-deletion. If a request is in a state of Pending Verification or On Hold, it won't get auto-deleted. 


### Email after deletion

Users will now receive an email after the deletion of their account and data.

<img width="459" alt="Screenshot 2021-12-01 at 8 56 15 PM" src="https://user-images.githubusercontent.com/31363128/144262698-1967b6df-7846-49b6-8b32-9472c923de9d.png">

### Steps to test:

1. Go to Website Settings. Scroll down to Account Deletion Settings.
2.  Enable Show Account Deletion Link in My Account Page.
3. Enter appropriate days for auto-deletion. Save
4. Go to the website. From the navbar go to the My Account Page.
5. You will find the link to the Request for Account Deletion form. Click on it.
6. On the form, verify the number of days. Enter the email id of the user whose account you want to delete. Submit.
7. Check emails. From the email, confirm the request.
8. From desk go to Personal Data Deletion request doctype. Find your request document there.
9. From the top right corner click on the button **Delete Data**. You should have a System Manager role for this.
10. Once the data is deleted, verify another email that you will receive.

### Documentation Draft:

https://docs.erpnext.com/docs/v13/user/manual/en/setting-up/personal-data-deletion/edit?wiki_page_patch=a0dd251ea9
